src/components/FlatTreeRenderer.js

```diff
@@ -0,0 +1,436 @@
+          x: `${xOffsetTo}rem`,
+          selectionOpacity: item.isCursor ? TEXT_SELECTION_OPCAITY : 0,
+          ...!isNaN(yOffsetTo) ? { y: `${yOffsetTo}px` } : {}
+        })
+      },
+      leave: item => async next => {
+        // we need new nodes height for animating out y offset animations
+        // waiting for heightObject update
+        await wait(20)
+
+        const isSecondColumn = item.viewInfo.table.column === 2
+
+        // find the item which is available in new flat array and was above the leaving node in old array
+        const lastItem = oldFlatArray.slice(0, item.index).reduce(
+          (acc, _item) => {
+            const isSibling = _item.path.length === item.path.length && isDescendant(contextOf(pathToContext(_item.path)), pathToContext(item.path))
+            return flatArrayKey[_item.key] && (isDescendant(pathToContext(_item.path), pathToContext(item.path)) || isSibling) ? _item : acc
+          },
+          null)
+
+        // if there such item then find it's new state item
+        const newLastItem = lastItem ? flatArrayKey[lastItem.key] : null
+
+        const xOffset = calculateXOffset(newLastItem || item, visibleStartDepth)
+
+        // right to left animation for second columns during unmount
+        const finalXOffset = xOffset - (isSecondColumn && !newLastItem ? 5 : 0)
+
+        const yOffsetObject = calculateYoffset({ ...heightObj.current }, flatArray)
+        // if node above is unmounting then this node needs to animate yoffset accordingly to prevent overlap with visible nodes
+        const heightAbove = newLastItem ? yOffsetObject[newLastItem.key] : yOffsetObjectOld[item.key]
+
+        await next({
+          x: `${finalXOffset}rem`,
+          ...!isNaN(heightAbove) ? { y: `${heightAbove}px` } : {},
+          opacity: 0
+        })
+      },
+      update: i => {
+        const item = flatArrayKey[i.key] || i
+        const heightAbove = yOffsetObject[i.key]
+        const xOffset = calculateXOffset(item, visibleStartDepth)
+
+        return {
+          opacity: item.isDistantThought ? DISTANT_THOUGHT_OPACITY : 1,
+          rotation: item.expanded ? 90 : 0,
+          selectionOpacity: item.isCursor ? TEXT_SELECTION_OPCAITY : 0,
+          x: `${xOffset}rem`,
+          ...isNaN(heightAbove) ? {} : { y: `${heightAbove}px` },
+        }
+      },
+    }
+  )
+
+  /**
+   * Debounced memoized function that updates height.
+   */
+  const debouncedHeightUpdate = useCallback(_.debounce(() => {
+    setHeightObject(heightObjectPendingRef.current)
+    heightObjectPendingRef.current = {}
+  }, 12), [])
+
+  /**
+   * Callback memoized function that runs on node height change.
+   */
+  const nodeHeightChangeHandler = useCallback(({ height, key }) => {
+    heightObjectPendingRef.current = { ...heightObjectPendingRef.current, [key]: height }
+    debouncedHeightUpdate()
+  }, [])
+
+  return (
+    <animated.div style={{ marginTop: '5rem', marginLeft: '5rem', height: '100%' }}>
```

Comment:
Can this be a regular `div` since it is not animating anything?

Reply:
Yes will do that